enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+    apiServerElb:
+      attributes: {}
+  ready: false
+```
+
+This manifest describes the network, load balancers, security group rules, etc. that will be used or provisioned for
+the target cluster. Control plane machines are provisioned by [AWSMachines][AWSMachines] & [CAPI Machines][CAPIMachines]
+manifests.
+
+##### Additional Infrastructure
+
+The Installer will generate any additional infrastructure that is needed but not handled by the CAPI provider, either
+because it is out of scope (e.g. IAM, DNS) or not adopted upstream (e.g. split-horizon load balancers). The Installer
+will provide hooks into the provisioning lifecycle that can be used to provision resources using direct SDK calls
+or other tooling.
+
+The AWS proof-of-concept implementation utilizes hooks defined in this interface:
+
+```go
+type CAPIInfraHelper interface {
+	// PreProvision is called before provisioning using CAPI controllers has begun.
+	// and should be used to create dependencies needed for CAPI provisioning,
+	// such as IAM roles or policies.
+	PreProvision(in PreProvisionInput) error
+
+	// ControlPlaneAvailable is called once cluster.Spec.ControlPlaneEndpoint.IsValid()
+	// returns true, typically after load balancers have been provisioned. It can be used
+	// to create DNS records.
+	ControlPlaneAvailable(in ControlPlaneAvailableInput) error
+}
+
+type PreProvisionInput struct{ clusterID string }
+type ControlPlaneAvailableInput struct{ *clusterv1.Cluster }
+```
+
+For AWS, IAM roles needed by the CAPA provider are created with `PreProvision` and DNS Records
+are created upon `ControlPlaneAvailable`. This interface would be implemented by each cloud provider
+and can be expanded as needed.
+
+##### Bootstrap Resources
+
+The Bootstrap Machine is created like the other control plane nodes, by defining a CAPI Machine resource.
+Accordingly, the bootstrap machine can be deleted by simply deleting the object. This works for all platforms:
+
+```go
+	if err := c.Delete(ctx, &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      capiutils.GenerateBoostrapMachineName(metadata.InfraID),
+			Namespace: capiutils.Namespace,
+		},
+	});
+```
+
+Further work needs to be done to determine how to delete additional bootstrap resources, such as SSH security group rules
+and the bootstrap ignition S3 bucket. These could be deleted either through updating the relevant manifests or created &
+deleted out-of-band using hooks.
+
+### Risks and Mitigations
+
+While we do not expect these changes to introduce a significant security risk, we are working with product security teams
+to ensure they are aware of the changes and are able to review.
+
+
+### Drawbacks
+
+By depending on CAPI providers whose codebases live in a repository external to the Installer,
+the process for developing features and delivering fixes is more complex than in a monolothic repo. 
+While the same could be true for the Installer Terraform dependency; the CAPI providers will 
```

Comment:
I believe we've resolved this for Terraform, so it won't in fact be true.

Reply:
Initially (until they can be pulled from the release image at build time)  the CAPI providers are vendored to the installer dir the same was as terraform providers. I've added details in the enhancement on the CAPI vendoring, which is equivalent to: https://github.com/openshift/installer/tree/master/terraform